consensus_encoding/tests/composition.rs

```diff
@@ -27,9 +27,27 @@ impl Encodable for CompositeData {
     }
 }
 
+/// A unified error type for [`CompositeDataDecoder`].
+#[derive(Debug, Clone, PartialEq, Eq)]
+enum CompositeError {
+    Eof(UnexpectedEof),
+}
+
+impl From<UnexpectedEof> for CompositeError {
+    fn from(eof: UnexpectedEof) -> Self { CompositeError::Eof(eof) }
+}
+
+impl core::fmt::Display for CompositeError {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            CompositeError::Eof(eof) => write!(f, "Error in first array: {}", eof),
```

Comment:
Its not necessarily the first array, right?

Reply:
Ah crap, yea, forgot to update this one.